backend/dataall/aws/handlers/lakeformation.py

```diff
@@ -6,33 +6,32 @@
 from .sts import SessionHelper
 
 log = logging.getLogger('aws:lakeformation')
-
+PIVOT_ROLE_NAME_PREFIX = "datallPivotRole"
 
 class LakeFormation:
     def __init__(self):
         pass
 
     @staticmethod
-    def describe_resource(resource_arn, role_arn, accountid, region):
+    def check_existing_lf_registered_location(resource_arn, accountid, region):
         """
-        Describes a LF data location
+        Checks if there is a non-dataall-created registered location for the Dataset
+        Returns False is already existing location else return the resource info
         """
         try:
             session = SessionHelper.remote_session(accountid)
             lf_client = session.client('lakeformation', region_name=region)
-
             response = lf_client.describe_resource(ResourceArn=resource_arn)
-
-            log.info(f'LF data location already registered: {response}, checking if data.all registered it ...')
-            if response['ResourceInfo']['RoleArn'] == role_arn:
+            registered_role_name = response['ResourceInfo']['RoleArn'].lstrip(f"arn:aws:iam::{accountid}:role/")
+            pivot_role_name = SessionHelper.get_delegation_role_name()
+            log.info(f'LF data location already registered: {response}, registered with role {registered_role_name}')
+            if registered_role_name[:14] == PIVOT_ROLE_NAME_PREFIX:
```

Comment:
This will always be false. (`registered_role_name[:14]` is `dataallPivotRo`)
Let's better use `registered_role_name.startswith(PIVOT_ROLE_NAME_PREFIX)` :)

Reply:
Yes, I don't know why I read that startswith was not available in 3.8 (our Lambdas Python version, yes we should upgrade them since Lambda Python 3.10 was released last week), but checking again it seems to be there, so I made the changes

Reply:
Thank you!!